src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -472,7 +496,7 @@ private static unsafe bool ParseNumber(ref char* str, char* strEnd, NumberStyles
             int digEnd = 0;
             while (true)
             {
-                if ((ch >= '0' && ch <= '9') || (((options & NumberStyles.AllowHexSpecifier) != 0) && ((ch >= 'a' && ch <= 'f') || (ch >= 'A' && ch <= 'F'))))
+                if (IsDigit(ch) || (((options & NumberStyles.AllowHexSpecifier) != 0) && ((ch >= 'a' && ch <= 'f') || (ch >= 'A' && ch <= 'F'))))
```

Comment:
Should we use the same trick from IsDigit to optimize the following checks?
`(ch >= 'a' && ch <= 'f')`
`(ch >= 'A' && ch <= 'F')`

Reply:
This code is going to be deleted in a PR I'll put up shortly.